Mathlib/Data/Multiset/DershowitzManna.lean

```diff
@@ -286,47 +228,28 @@ private lemma transLT_of_isDershowitzMannaLT [dec : DecidableEq α]
         simp
       rcases this with ⟨z,z_in_Z⟩
       let newZ := Z.erase z
-      have newZ_nonEmpty : newZ ≠ ∅ := by
-        have : ∀ (n : ℕ), n ≠ 0 → n ≠ 1 → n ≥ 2 := by
-          intros n h0 h1
-          cases n
-          case zero => contradiction
-          case succ m =>
-            cases m
-            case zero => contradiction
-            case succ n =>
-              apply Nat.succ_le_succ
-              simp_all
-        have : 0 < card (erase Z z) := by
-          subst M_def N_def
-          simp_all only
-          [empty_eq_zero, ne_eq, card_eq_zero, not_false_eq_true, ge_iff_le, card_erase_of_mem,
-          Nat.pred_eq_sub_one, tsub_pos_iff_lt]
-          apply this
-          · simp_all only [card_eq_zero, not_false_eq_true]
-          · simp_all only [not_false_eq_true]
-        rw [card_pos] at this
-        simp_all only [empty_eq_zero, ne_eq, not_false_eq_true]
+      have newZ_nonEmpty : newZ ≠ 0 := by
+        simp [newZ, ← sub_singleton, tsub_eq_zero_iff_le]
+        aesop
```

Comment:
Thanks for the golfing! I am wondering here: Is it okay to leave `aesop` in for mathlib PRs and actually preferred in order to have shorter/readable proofs? In previous rounds we tried to always replace it by explicit proofs, but they can be lengthy :thinking:

Reply:
Oh no, using aesop is perfectly so long as performance is not too terrible